File: frontend/src/types/mahasiswa.ts
export type Mahasiswa = {
  id: number;
  npm: string;
  nama: string;
  kelas: string;
  profile: string;
};

export type ListResponse<T> = {
  data: T[];
  total: number;
  page: number;
  limit: number;
};

export type SingleResponse<T> = { data: T };

File: backend/internal/models/mahasiswa.go
package models

// Sesuai tabel `mahasiswa` (id, npm, nama, kelas, minat, profile)
type Mahasiswa struct {
	ID      uint   `gorm:"primaryKey" json:"id"`
	NPM     string `json:"npm"`
	Nama    string `json:"nama"`
	Kelas   string `json:"kelas"`
	Minat   string `json:"minat"`
	Profile string `json:"profile"`
}

func (Mahasiswa) TableName() string { return "mahasiswa" }
